src/coreclr/src/jit/emit.cpp

```diff
@@ -7692,7 +7692,12 @@ regMaskTP emitter::emitGetGCRegsKilledByNoGCCall(CorInfoHelpFunc helper)
             break;
 
         case CORINFO_HELP_PROF_FCN_LEAVE:
+#if defined(TARGET_ARM)
+            // profiler scratch remains gc live
+            result = RBM_PROFILER_LEAVE_TRASH & ~RBM_PROFILER_RET_SCRATCH;
+#else
```

Comment:
Can you please explain why this is only needed for `TARGET_ARM` and not other architectures?

Reply:
Arm is the only target that does this sort of return value sheltering in jitted code around the profiler leave hook.

No other architecture defines `RBM_PROFILER_RET_SCRATCH`.